src/code/code_800EC960.c

```diff
@@ -3809,48 +4307,62 @@ s32 func_800F5A58(u8 arg0) {
     }
 }
 
+/**
+ * Plays a sequence on the main bgm player, but stores the previous sequence to return to later
+ * Designed for the mini-boss sequence, but also used by mini-game 2 sequence
+ */
 void func_800F5ACC(u16 seqId) {
-    u16 temp_v0;
+    u16 curSeqId = func_800FA0B4(SEQ_PLAYER_BGM_MAIN);
 
-    temp_v0 = func_800FA0B4(SEQ_PLAYER_BGM_MAIN);
-    if ((temp_v0 & 0xFF) != NA_BGM_GANON_TOWER && (temp_v0 & 0xFF) != NA_BGM_ESCAPE && temp_v0 != seqId) {
+    if ((curSeqId & 0xFF) != NA_BGM_GANON_TOWER && (curSeqId & 0xFF) != NA_BGM_ESCAPE && curSeqId != seqId) {
         func_800F5E90(3);
-        if (temp_v0 != NA_BGM_DISABLED) {
-            D_80130628 = temp_v0;
+        if (curSeqId != NA_BGM_DISABLED) {
+            sPrevMainBgmSeqId = curSeqId;
         } else {
             osSyncPrintf("Middle Boss BGM Start not stack \n");
         }
+
         Audio_StartSeq(SEQ_PLAYER_BGM_MAIN, 0, seqId);
     }
 }
 
+/**
+ * Restores the previous sequence to the main bgm player before func_800F5ACC was called
+ */
 void func_800F5B58(void) {
-    if ((func_800FA0B4(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED) && (D_80130628 != NA_BGM_DISABLED) &&
+    if ((func_800FA0B4(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED) && (sPrevMainBgmSeqId != NA_BGM_DISABLED) &&
```

Comment:
does it make sense that this also checks there is a bgm currently playing? (if I'm reading this right)
\+ whatever `D_80130658` is for

Reply:
`func_800FA0B4` is indeed called `Audio_GetActiveSequence` and checks for the current sequence being played on the audio player given to as an argument. In keeping with the spirit of small PRs, while this is already documented locally, I will be naming this when I submit a PR documenting `code_800F9280.c` which deals with processing sequence commands and manages the "Active Sequences".

Along those same lines, `D_80130658` is actually called `gActiveSequences` (or `gActiveSeqs` for short, haven't finalized the name yet). This will also be named when documenting  `code_800F9280.c`. You can check out this branch if you're curious:

https://github.com/engineer124/oot/blob/cef74c8ae7647d64e54bb781d136290c9255f1af/include/z64audio.h#L941

and really this entire file:

https://github.com/engineer124/oot/blob/cef74c8ae7647d64e54bb781d136290c9255f1af/src/code/code_800F9280.c#L771

Reply:
Do we have a better name for this particular file?

Reply:
I’m going to assume you mean `code_800F9280.c` from this comment thread, instead of `code_800EC960.c` which is where this particular change and most other changes from this PR occur.

I have a few ideas for `code_800F9280.c` but nothing has really stuck yet. In MM, I’ve been trialling the name `audio_seq_processor.c` on my local branch, but it’s not quite doing it for me since `audio_seqplayer.c` could also be considered “processing sequences” as that is responsible for interpreting the music macro language. 

In summary, `code_800F9280.c` is sequence processing at the highest level. Any requests to start/stop/modify active sequences are managed through this file, primarily through “sequence commands” or `audio_seqcmd`s for short. This high level commands are available for any actor or code to use throughout the repo, but there are common, generic combinations of commands that are usually grouped together in `code_800EC960.c` that actors tend to use as well. (common things such as `Audio_PlayFanfare` need a combination of commands to turn on and off the right sequence players and set the right properties, so that’s more common than calling all the right commands directly).

So other names I’ve been thinking of are around the command system. Names like `audio_seq_commands.c` or `audio_seqcmd_processor.c` as examples.

Another thing that must be addressed is an entirely separate and independent set of commands exists further down in audio processing to transfer internal audio requests from the main thread to internal audio processing on the audio thread. This is the one of the primary things accomplished in `code_800E4FE0.c`. Currently, those different set of commands are simply called `Audio Commands` or `Audio_cmd`, which is not very distinguishable from `Audio_seqcmd`. So there needs to be discussion and agreement on how to distinguish those two types of commands. And how those commands are named might have an impact on the filename for `code_800F9280.c`

Alternatively, the name could emphasize the `gActiveSequences`/`gActiveSeq` aspect of this file, which is what the commands modify and update before the right info gets sent to the lower level “Audio_cmd”. Basing it on that would give a filename `audio_active_sequences`, but I’m not sure that name does it for me either.

Reply:
There's also a parallel between the files `code_800F9280.c` which is processing sequences at the highest level, and `code_800F7260.c` which is processing sfx at the highest level. It's not quite a true parallel as sfx's are a subset of sequences (sfx = seq 0), and the way they are handled are slightly different to ( `code_800F9280.c` deals with sequence commands and just active sequences, while `code_800F7260.c` deals with separate "sfx requests and active sfxs" and the transfer of information that way).

That's why I've been trialing `audio_sfx_processor` in MM to parallel my trial for `audio_seq_processor`, but again the name isn't quite sticking with me since sfx's are processed in many different files